scripts/jest/test-framework-setup.js

```diff
@@ -42,22 +34,41 @@ if (process.env.REACT_CLASS_EQUIVALENCE_TEST) {
       },
       toNotHaveBeenCalled() {
         return {
-          compare(actual) {
+          compare(actual, methodName) {
             return {
-              pass: callCount === 0,
+              pass: actual.__callCount === 0,
               message: () =>
-                'Expected test not to warn. If the warning is expected, mock ' +
-                "it out using spyOn(console, 'error'); and test that the " +
+                'Expected test not to call console.' +
+                methodName +
+                '(). ' +
+                'If the warning is expected, mock it out using ' +
+                "spyOn(console, '" +
+                methodName +
+                "') and test that the " +
                 'warning occurs.',
             };
           },
         };
       },
     });
   });
-  env.afterEach(() => {
-    expect(console.error).toBeReset();
-    expect(console.error).toNotHaveBeenCalled();
+
+  ['error', 'warn'].forEach(methodName => {
+    var oldMethod = console[methodName];
+    var newMethod = function() {
+      newMethod.__callCount++;
+      oldMethod.apply(this, arguments);
+    };
+    newMethod.__callCount = 0;
+    console[methodName] = newMethod;
+
+    env.beforeEach(() => {
+      newMethod.__callCount = 0;
+    });
+    env.afterEach(() => {
+      expect(console[methodName]).toBeReset(methodName);
```

Comment:
I think Jest has been changed since I wrote this so that instead of defining the two matchers you can just do

```
if (console[methodName] !== newMethod && !isSpy(console[methodName]) {
  throw ...
}
if (newMethod.__callCount !== 0) {
  throw ...
}
```

here.

Might be simpler.